client/src/routes/dashboard.svelte

```diff
@@ -7,29 +7,88 @@
 				redirect: '/login',
 			};
 		}
-		const res = await getExpenses({ userId: session.user.id });
+		const currencyTypes = (await getCurrencyTypes()).data;
 		return {
-			status: res.status,
 			props: {
-				expenses: res.data,
+				userId: session.user.id,
+				currencyTypes,
 			},
 		};
 	};
 </script>
 
 <script lang="ts">
-	import { Expense } from '$mock/api/models/expense_model';
+	import { onMount } from 'svelte';
+	import { CurrencyType, Expense, NewTransaction } from '$mock/api/models/expense_model';
 
 	import Drawer, { AppContent, Content } from '@smui/drawer';
 	import List, { Item, Text } from '@smui/list';
-	import { getExpenses } from '$api/dashboard';
+	import TransactionList from '$lib/components/TransactionList.svelte';
+	import Separator from '@smui/list/src/Separator.svelte';
+	import Button from '@smui/button/src/Button.svelte';
+	import NewTransactionDialog from '$lib/components/NewTransactionDialog.svelte';
+	import { getCurrencyTypes } from './api/currency';
 
-	export let expenses: Expense[] = [];
+	export let userId: number;
+	export let currencyTypes: CurrencyType[] = [];
+
+	let expenses: Expense[] = [];
+	let clickedExpense: Expense | null;
+	let newTransactionDialog: NewTransactionDialog;
+
+	onMount(() => {
+		fetch(`/api/expense/${userId}`, {
+			method: 'GET',
+			headers: {
+				'Content-Type': 'application/json',
+			},
+		}).then((res) => {
+			res.json().then((res) => {
+				expenses = res.data.expenses as Expense[];
+				clickedExpense = expenses && expenses[0] ? expenses[0] : null;
```

Comment:
Do we want to select the first expense by default (or leave it blank until the user selects one)?
If not, then this is not necessary anymore.
$: clickedExpense = .. line does the trick

Reply:
I figured, the first should always be showing, it's only a minor tweak to make if necessary, let's leave it like this for now.
Nonetheless I streamlined the promise-then forest that was present there with the much sleeker async design.